src/Documentation/Core-Features/Grain-Persistence.md

```diff
@@ -36,72 +36,184 @@ public class MyGrain<MyGrainState> ...
 }
 ```
 
-The Orleans Provider Manager framework provides a mechanism to specify & register different storage providers and storage options in the silo config file.
-
-```xml
-<OrleansConfiguration xmlns="urn:orleans">
-    <Globals>
-    <StorageProviders>
-        <Provider Type="Orleans.Storage.MemoryStorage" Name="DevStore" />
-        <Provider Type="Orleans.Storage.AzureTableStorage" Name="store1"
-            DataConnectionString="DefaultEndpointsProtocol=https;AccountName=data1;AccountKey=SOMETHING1" />
-        <Provider Type="Orleans.Storage.AzureBlobStorage" Name="store2"
-            DataConnectionString="DefaultEndpointsProtocol=https;AccountName=data2;AccountKey=SOMETHING2"  />
-    </StorageProviders>
+The Orleans framework provides a mechanism to specify & register different storage providers and configure them using `ISiloHostBuilder`,
+
+``` csharp
+var silo = new SiloHostBuilder()
+          .AddMemoryGrainStorage("DevStore")
+          .AddAzureTableGrainStorage("store1", options=>options.ConnectionString = "DefaultEndpointsProtocol=https;AccountName=data1;AccountKey=SOMETHING1")
+          .AddAzureBlobGrainStorage("store2", options=>options.ConnectionString = "DefaultEndpointsProtocol=https;AccountName=data2;AccountKey=SOMETHING2")
+          .Builder();
 ```
 
-## Configuring Storage Providers
+## Configuring IGrainStorage
```

Comment:
Could you add a small introduction phrase? Just to list quickly all providers before